src/pages/Profile/profileSheet.tsx

```diff
@@ -1,65 +1,75 @@
-import {useLogoutMutation} from "../../redux/slices/authSlice";
-import {useNavigate} from "react-router-dom";
+import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar';
+import {
+  Sheet,
+  SheetTrigger,
+  SheetContent,
+  SheetHeader,
+  SheetTitle,
+  SheetDescription,
+  SheetFooter,
+  SheetClose,
+} from '@/components/ui/sheet';
+import { Button } from '@/components/ui/button';
+import { Input } from '@/components/ui/input';
+import { Label } from '@/components/ui/label';
+import { useLogoutMutation } from '../../redux/api/authApi.ts';
+import { useNavigate } from 'react-router-dom';
 
 const ProfileSheet: FC = () => {
-    const [logout] = useLogoutMutation();
-    const navigate = useNavigate();
+  const [logout] = useLogoutMutation();
+  const navigate = useNavigate();
 
-    const handleLogout = async () => {
-        try {
-            await logout({}).unwrap();
-            navigate('/login');
-            console.log("Logged out");
-        } catch (error) {
-            console.error("Logout failed", error);
-        }
-    };
-    return (
-        <Sheet>
-            <SheetTrigger asChild>
-                <Avatar className="cursor-pointer">
-                    <AvatarImage src="https://github.com/shadcn.png" />
-                    <AvatarFallback>CN</AvatarFallback>
-                </Avatar>
-            </SheetTrigger>
-            <SheetContent>
-                <SheetHeader>
-                    <SheetTitle>Edit profile</SheetTitle>
-                    <SheetDescription>
-                        Make changes to your profile here. Click save when you're done.
-                    </SheetDescription>
-                </SheetHeader>
-                <div className="grid gap-4 py-4">
-                    <div className="grid grid-cols-4 items-center gap-4">
-                        <Label htmlFor="name" className="text-right">
-                            Name
-                        </Label>
-                        <Input id="name" value="Pedro Duarte" className="col-span-3" />
-                    </div>
-                    <div className="grid grid-cols-4 items-center gap-4">
-                        <Label htmlFor="username" className="text-right">
-                            Username
-                        </Label>
-                        <Input id="username" value="@peduarte" className="col-span-3" />
-                    </div>
-                </div>
-                <SheetFooter>
-                    <SheetClose asChild>
-                        <Button type="submit">Save changes</Button>
-                    </SheetClose>
-                    <Button variant="destructive" onClick={handleLogout}>
-                        Log Out
-                    </Button>
-                </SheetFooter>
-            </SheetContent>
-        </Sheet>
-    );
+  const handleLogout = async () => {
+    try {
+      await logout({}).unwrap();
+
+      navigate('/login');
```

Comment:
Asi jsem už psal předtím, ale navigovaní po přihlašení by se neměla / nemusela dělat takhle. Ale namísto toho pro anonymous users v router mít jen login screen, aby jiné neměl dostupné (plus ješte 404 page). To zabezpečí, že po odhlašení ho to automaticky přesměruje na dostupnou screen a to bude ta login. A ješte nebudeš muset pak řešit na každé screen handlovaní prihlašený / neprihlašený. Ale vidím, že si to řešil přes protected route wrapper, dá se i tak...